File: data/code_pairs/fortran/1988.f90
module data_gllmesh
  implicit none
  ! Assuming dp kind parameter for double precision
  integer, parameter :: dp = selected_real_kind(15, 307)
  public
  real(kind=dp), dimension(:,:,:), allocatable :: sgll, zgll
  real(kind=dp), dimension(:,:,:), allocatable :: sgll_fluid, zgll_fluid
  real(kind=dp), dimension(:,:,:), allocatable :: sgll_solid, zgll_solid

  real(kind=dp) :: hmin_glob, hmax_glob
  real(kind=dp) :: min_distance_dim
  real(kind=dp) :: min_distance_nondim

  real(kind=dp) :: char_time_max
  integer :: char_time_max_globel
  real(kind=dp) :: char_time_max_rad, char_time_max_theta
  real(kind=dp) :: char_time_min
  integer :: char_time_min_globel
  real(kind=dp) :: char_time_min_rad, char_time_min_theta
end module data_gllmesh

File: data/code_pairs/cpp/1988.cpp
#ifndef DATA_GLLMESH_H
#define DATA_GLLMESH_H

#include <vector>

class DataGllMesh {
public:
    std::vector<std::vector<std::vector<double>>> sgll, zgll;
    std::vector<std::vector<std::vector<double>>> sgll_fluid, zgll_fluid;
    std::vector<std::vector<std::vector<double>>> sgll_solid, zgll_solid;

    double hmin_glob, hmax_glob;
    double min_distance_dim;
    double min_distance_nondim;

    double char_time_max;
    int char_time_max_globel;
    double char_time_max_rad, char_time_max_theta;
    double char_time_min;
    int char_time_min_globel;
    double char_time_min_rad, char_time_min_theta;

    DataGllMesh() = default;
    ~DataGllMesh() = default;

    void allocateSgll(int x, int y, int z) {
        sgll.resize(x, std::vector<std::vector<double>>(y, std::vector<double>(z)));
    }

    // Similar allocation functions can be defined for zgll, sgll_fluid, etc.
};

#endif // DATA_GLLMESH_H